hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerProtocolServerSideTranslatorPB.java

```diff
@@ -143,67 +147,67 @@ public OMResponse submitRequest(RpcController controller,
       throw new ServiceException(e);
     }
 
-    OMResponse response = 
-        dispatcher.processRequest(validatedRequest, this::processRequest,
-        request.getCmdType(), request.getTraceID());
-    
+    OMResponse response = dispatcher.processRequest(validatedRequest,
+        this::processRequest,
+        request.getCmdType(),
+        request.getTraceID());
+
     return requestValidations.validateResponse(request, response);
   }
 
-  private OMResponse processRequest(OMRequest request) throws
-      ServiceException {
+  private OMResponse processRequest(OMRequest request) throws ServiceException {
     OMClientRequest omClientRequest = null;
-
     boolean s3Auth = false;
+
     try {
+      if (request.getCmdType() == CreateSnapshot) {
```

Comment:
Its looks little odd to have a snapshot specific hook here. Either 
- you can make it a separate function that does getCmdType based processing e.g. (add the snapshotId for snapshot create request), or
-  you can pass the unique snapshot Id all the way from the client.

Reply:
Initially I thought about passing `SnapshotID` from client. Then I released that clients can override it and pass non-unique (random/same) SnapshotID. Let me know if clients can't override it or are not supposed to do so. If so, we can pass `SnapshotID` from client.  Otherwise I'll create separate function for it.

Reply:
Moved `getCmdType` based request mutation to `RequestValidations` because we already clone request there before submitting it to Ratis.